money-core/src/main/scala/com/comcast/money/core/internal/SpanLocal.scala

```diff
@@ -46,36 +46,33 @@ object SpanLocal extends SpanContext {
   override def current: Option[Span] = Option(threadLocalCtx.get).flatMap(_.headOption)
 
   override def push(span: Span): Unit =
-    if (span != null)
+    if (span != null) {
       Option(threadLocalCtx.get) match {
-        case Some(stack) =>
-          stack.push(span)
-          setSpanMDC(Some(span.info.id))
-          setSpanNameMDC(Some(span.info.name))
+        case Some(current) =>
+          threadLocalCtx.set(span :: current)
         case None =>
-          threadLocalCtx.set(new Stack[Span]())
-          push(span)
+          threadLocalCtx.set(List(span))
       }
+      setSpanMDC(Some(span.info.id))
+      setSpanNameMDC(Some(span.info.name))
+    }
 
   override def pop(): Option[Span] =
-    Option(threadLocalCtx.get).map {
-      stack =>
-        // remove the current span in scope for this thread
-        val spanId = stack.pop()
-
-        // rolls back the mdc span to the parent if present, if none then it will be cleared
-        val head = stack.headOption
+    threadLocalCtx.get match {
+      case span :: rest =>
+        threadLocalCtx.set(rest)
+        val head = rest.headOption
         setSpanMDC(head.map(_.info.id))
         setSpanNameMDC(head.map(_.info.name))
-
-        spanId
+        Some(span)
+      case _ => None
     }
 
   /**
    * Clears the entire call stack for the thread
    */
-  override def clear() = {
-    if (threadLocalCtx != null) threadLocalCtx.remove()
+  override def clear(): Unit = {
+    threadLocalCtx.remove()
```

Comment:
unclear on why we removed the threadLocalCtx != null check.  it seemingly doesn't make sense that it is there, but not remembering why it was there in the first place leads me to believe we should just leave it.

Reply:
I'm not sure under what circumstance that field could be `null`.  Maybe the intention was to check if `threadLocalCtx.get` was `null`?

I can add the `null` check back if it makes you feel more comfortable, but that begs the question as to why we're not checking that in the other methods.